ml/write_dataset_config.py

```diff
@@ -71,636 +75,180 @@ def main(args):
-        classes_map = {
-            "ggH": "ggh",
-            "qqH": "qqh",
-            "ZTT": "ztt",
-            "EMB": "ztt",
-            "ZL": "misc",
-            "TTT": "tt",
-            "TTL": "tt",
-            "W": "misc",
-            "VVT": "db",
-            "VVL": "db",
-        }
-        for estimation in [
-                ggHEstimation("ggH", era, args.base_path, channel),
-                qqHEstimation("qqH", era, args.base_path, channel),
-                ZTTEstimation(era, args.base_path, channel),
-                ZLEstimation(era, args.base_path, channel),
-                TTTEstimation(era, args.base_path, channel),
-                TTLEstimation(era, args.base_path, channel),
-                WEstimation(era, args.base_path, channel),
-                VVLEstimation(era, args.base_path, channel),
-                VVTEstimation(era, args.base_path, channel),
-        ]:
-            output_config["processes"][estimation.name] = {
-                "files": [
-                    str(f).replace(args.base_path.rstrip("/") + "/", "")
-                    for f in estimation.get_files()
-                ],
-                "cut_string": (estimation.get_cuts() + channel.cuts +
-                               additional_cuts).expand(),
-                "weight_string":
-                estimation.get_weights().extract(),
-                "class":
-                classes_map[estimation.name]
-            }
-
-        # Same sign selection for data-driven QCD
-        estimation = DataEstimation(era, args.base_path, channel)
-        estimation.name = "QCD"
-        channel_ss = copy.deepcopy(channel)
-        channel_ss.cuts.get("os").invert()
-        output_config["processes"][estimation.name] = {
-            "files": [
-                str(f).replace(args.base_path.rstrip("/") + "/", "")
-                for f in estimation.get_files()
-            ],
-            "cut_string": (estimation.get_cuts() + channel_ss.cuts +
-                           additional_cuts).expand(),
-            "weight_string":
-            estimation.get_weights().extract(),
-            "class":
-            "ss"
-        }
-
-    ############################################################################
-
+    ###
+    # Same sign selection for data-driven QCD
+    estimation = DataEstimation(era, args.base_path, channel)
+    estimation.name = "QCD"
+    channel_qcd = copy.deepcopy(channel)
+
+    if args.channel != "tt":
+        ## os= opposite sign
+        channel_qcd.cuts.get("os").invert()
+    # Same sign selection for data-driven QCD
+    else:
+        channel_qcd.cuts.remove("tau_2_iso")
+        channel_qcd.cuts.add(
+            Cut("byTightIsolationMVArun2v1DBoldDMwLT_2<0.5", "tau_2_iso"))
+        if args.era=="2016":
+            channel_qcd.cuts.add(
+                Cut("byLooseIsolationMVArun2v1DBoldDMwLT_2>0.5", "tau_2_iso_loose"))
+        elif args.era=="2017":
```

Comment:
For the latest implementation, you don't need to distinguish here between years (see commend above)